internal/storage/bucket_handle_test.go

```diff
@@ -138,3 +138,29 @@ func (t *BucketHandleTest) TestNewReaderMethodWithInvalidGeneration() {
 	AssertNe(nil, err)
 	AssertEq(nil, rc)
 }
+
+func (t *BucketHandleTest) TestDeleteObjectMethodWithValidObject() {
+	error := t.bucketHandle.DeleteObject(context.Background(),
+		&gcs.DeleteObjectRequest{
+			Name:                       TestObjectName,
+			Generation:                 0,
+			MetaGenerationPrecondition: nil,
+		})
+
+	AssertEq(nil, error)
+}
+
+func (t *BucketHandleTest) TestDeleteObjectMethodWithMissingObject() {
+	error := t.bucketHandle.DeleteObject(context.Background(),
+		&gcs.DeleteObjectRequest{
+			Name:                       missingObjectName,
+			Generation:                 0,
+			MetaGenerationPrecondition: nil,
+		})
+
+	AssertEq("storage: object doesn't exist", error.Error())
+}
+
+// Delete-method of the fake server object is not returning an error for invalid Generation and MetaGenerationPrecondition.
```

Comment:
Instead i would suggest rephrasing it to FakeGCSServer is not handling generation and metageneration checks for Delete flow. 
And add this comment at line no: 27.

also link looks weird. Use: https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/vendor/github.com/fsouza/fake-gcs-server/fakestorage/object.go#L515

Reply:
fixed